functions/pipeline/shared/db_access/db_access_v2.py

```diff
@@ -155,6 +152,98 @@ def get_tag_complete_images(self, number_of_images, user_id):
             conn.close()
         return tag_complete_images
 
+    def get_ready_to_tag_images(self, number_of_images, user_id):
+        if number_of_images <= 0:
+            raise ArgumentException("Parameter must be greater than zero")
+        ready_to_tag_images = {}
+        try:
+            conn = self._db_provider.get_connection()
+            try:
+                cursor = conn.cursor()
+                query = ("SELECT b.ImageId, b.ImageLocation, a.TagStateId FROM Image_Tagging_State a "
+                        "JOIN Image_Info b ON a.ImageId = b.ImageId WHERE a.TagStateId = {1} order by "
+                        "a.createddtim DESC limit {0}")
+                cursor.execute(query.format(number_of_images, ImageTagState.READY_TO_TAG))
+                for row in cursor:
+                    logging.debug('Image Id: {0} \t\tImage Name: {1} \t\tTag State: {2}'.format(row[0], row[1], row[2]))
+                    ready_to_tag_images[row[0]] = str(row[1])
+            finally:
+                cursor.close()
+        except Exception as e:
+            logging.error("An errors occured getting ready to tag images: {0}".format(e))
+            raise
+        finally:
+            conn.close()
+        return ready_to_tag_images
+    
+
+    def get_image_info_for_image_ids(self, image_ids):
+        try:
+            conn = self._db_provider.get_connection()
+            try:
+                cursor = conn.cursor()
+                ids = ''
+                for id in image_ids:
+                    ids += str(id) + ','
+                ids = ids[:-1]
+                query = ("select imageid, originalimagename, imagelocation, height, width, createdbyuser from image_info where imageid IN ({0});")
+                cursor.execute(query.format(ids))
+                logging.debug("Got image info back for image_id={}".format(image_ids))
+
+                images_info = []
+                for row in cursor:
+                    info = {}
+                    info['height'] = row[3]
```

Comment:
@samiyaakhtar I know it's faster create a list of dictionaries but I think creating a Class that defined the properties would allow you to share code on the backend (Data access and function) with the frontend (CLI client). This would create a contract. You can deserialize the JSON to a Class and serialize a Class to JSON. See an example here: https://stackoverflow.com/a/15477213

Something to thing about for the future